solidity-plugin/src/main/java/org/sonarsource/solidity/MetricsVisitor.java

```diff
@@ -0,0 +1,189 @@
+package org.sonarsource.solidity;
+
+import java.util.Set;
+import org.antlr.v4.runtime.Token;
+import org.antlr.v4.runtime.tree.TerminalNode;
+import org.sonarsource.solidity.frontend.SolidityBaseVisitor;
+import org.sonarsource.solidity.frontend.SolidityParser;
+import org.sonarsource.solidity.frontend.SolidityParser.BreakStatementContext;
+import org.sonarsource.solidity.frontend.SolidityParser.ContinueStatementContext;
+import org.sonarsource.solidity.frontend.SolidityParser.ContractDefinitionContext;
+import org.sonarsource.solidity.frontend.SolidityParser.ContractPartContext;
+import org.sonarsource.solidity.frontend.SolidityParser.DoWhileStatementContext;
+import org.sonarsource.solidity.frontend.SolidityParser.EmitStatementContext;
+import org.sonarsource.solidity.frontend.SolidityParser.ExpressionStatementContext;
+import org.sonarsource.solidity.frontend.SolidityParser.ForStatementContext;
+import org.sonarsource.solidity.frontend.SolidityParser.FunctionDefinitionContext;
+import org.sonarsource.solidity.frontend.SolidityParser.IfStatementContext;
+import org.sonarsource.solidity.frontend.SolidityParser.ReturnStatementContext;
+import org.sonarsource.solidity.frontend.SolidityParser.ThrowStatementContext;
+import org.sonarsource.solidity.frontend.SolidityParser.VariableDeclarationStatementContext;
+import org.sonarsource.solidity.frontend.SolidityParser.WhileStatementContext;
+
+public class MetricsVisitor extends SolidityBaseVisitor<Token> {
+  protected FileMeasures fileMeasures;
+  int functionCounter = 0;
+  int statements = 0;
+  int contractCounter = 0;
+
+  public MetricsVisitor(SolidityParser parser) {
+    this.fileMeasures = new FileMeasures();
+    parser.sourceUnit().accept(this);
+    this.fileMeasures.setStatementNumber(statements);
+    this.fileMeasures.setFunctionNumber(functionCounter);
+    this.fileMeasures.setContractNumber(contractCounter);
+    this.fileMeasures.setCommentLinesNumber(computeLinesOfComments(parser.comments));
+    TerminalNode node = parser.sourceUnit().EOF();
+    this.fileMeasures.setLinesOfCodeNumber(node.getSymbol().getLine());
+  }
+
+  private int computeLinesOfComments(Set<Token> tokens) {
+    int comments = 0;
+    for (Token token : tokens) {
+      if (token.getType() == 118) {
+        comments++;
+      } else {
+        comments += token.getText().split("\r\n|\r|\n").length;
+      }
+    }
+    return comments;
+  }
+
+  // @Override
+  // public Token visitSourceUnit(SourceUnitContext ctx) {
+  // List<FunctionDefinitionContext> functionList = contractPartList
+  // .stream()
+  // .map(x -> x.functionDefinition())
+  // .filter(Objects::nonNull)
+  // .collect(Collectors.toList());
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: This block of commented-out lines of code should be removed. ([squid:CommentedOutCodeLine](https://sonarcloud.io/organizations/stylianos-agapiou-sonarsource-gi/rules#rule_key=squid%3ACommentedOutCodeLine))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=org.sonarsource.solidity%3Asonar-solidity&pullRequest=16&issues=AWQTGK7hbXXTsx7O4tIH&open=AWQTGK7hbXXTsx7O4tIH)

[ISSUE_KEY]: AWQTGK7hbXXTsx7O4tIH